patches/MCLOUD-3025__sitemap_generation_warnings__2.3.3.patch

```diff
@@ -0,0 +1,84 @@
+diff -Nuar a/vendor/magento/module-sitemap/Model/Observer.php b/vendor/magento/module-sitemap/Model/Observer.php
+--- a/vendor/magento/module-sitemap/Model/Observer.php
++++ b/vendor/magento/module-sitemap/Model/Observer.php
+@@ -7,8 +7,10 @@ namespace Magento\Sitemap\Model;
+
+ use Magento\Sitemap\Model\EmailNotification as SitemapEmail;
+ use Magento\Framework\App\Config\ScopeConfigInterface;
++use Magento\Framework\App\ObjectManager;
+ use Magento\Sitemap\Model\ResourceModel\Sitemap\CollectionFactory;
+ use Magento\Store\Model\ScopeInterface;
++use Psr\Log\LoggerInterface;
+
+ /**
+  * Sitemap module observer
+@@ -61,20 +63,33 @@ class Observer
+      */
+     private $emailNotification;
+
++    /**
++     * @var int
++     */
++    private $retryInterval = 10;
++
++    /**
++     * @var LoggerInterface
++     */
++    private $logger;
++
+     /**
+      * Observer constructor.
+      * @param ScopeConfigInterface $scopeConfig
+      * @param CollectionFactory $collectionFactory
+      * @param EmailNotification $emailNotification
++     * @param LoggerInterface|null $logger
+      */
+     public function __construct(
+         ScopeConfigInterface $scopeConfig,
+         CollectionFactory $collectionFactory,
+-        SitemapEmail $emailNotification
++        SitemapEmail $emailNotification,
++        LoggerInterface $logger = null
+     ) {
+         $this->scopeConfig = $scopeConfig;
+         $this->collectionFactory = $collectionFactory;
+         $this->emailNotification = $emailNotification;
++        $this->logger = $logger ?: ObjectManager::getInstance()->get(LoggerInterface::class);
+     }
+
+     /**
+@@ -87,6 +102,7 @@ class Observer
+     public function scheduledGenerateSitemaps()
+     {
+         $errors = [];
++        $sitemapsWithError = [];
+         $recipient = $this->scopeConfig->getValue(
+             Observer::XML_PATH_ERROR_RECIPIENT,
+             ScopeInterface::SCOPE_STORE
+@@ -107,11 +123,25 @@ class Observer
+             try {
+                 $sitemap->generateXml();
+             } catch (\Exception $e) {
++                $sitemapsWithError[] = $sitemap;
+                 $errors[] = $e->getMessage();
+             }
+         }
++
+         if ($errors && $recipient) {
+-            $this->emailNotification->sendErrors($errors);
++            sleep($this->retryInterval);
++            $this->logger->debug('Sitemap generation retry occurred');
```

Comment:
I would log the error code itself too. and then that we retry generation.
And at the end add error to logs it is still present (that one which customer will receive)

Reply:
Improved logging implemented